crates/astria-composer/src/ds/streaming_client.rs

```diff
@@ -0,0 +1,46 @@
+use async_trait::async_trait;
+use ethers::providers::{
+    JsonRpcClient, Middleware, Provider as EthersProvider, PubsubClient, StreamExt,
+};
+use tokio::sync::mpsc as tokio_mpsc;
+use color_eyre::eyre::{self, Context};
+
+use super::{RollupChainId, RollupTx, RollupTxExt};
+
+#[async_trait]
+pub(crate) trait StreamingClient {
+    type Error;
+    async fn start_stream(
+        &self,
+        chain_id: RollupChainId,
+    ) -> Result<tokio_mpsc::UnboundedReceiver<RollupTxExt>, Self::Error>;
+}
+
+#[async_trait]
+impl<C> StreamingClient for EthersProvider<C>
```

Comment:
why are we implementing this for the `EthersProvider` type rather than our own `EthProvider`?

Reply:
See above